test/extensions/filters/http/common/compressor/compressor_filter_test.cc

```diff
@@ -233,6 +238,82 @@ TEST_F(CompressorFilterTest, CompressRequest) {
   doResponseNoCompression(headers);
 }
 
+TEST_F(CompressorFilterTest, CompressRequestNoContentLength) {
+  setUpFilter(R"EOF(
+{
+  "request_direction_config": {},
+  "compressor_library": {
+     "name": "test",
+     "typed_config": {
+       "@type": "type.googleapis.com/envoy.extensions.compression.gzip.compressor.v3.Gzip"
+     }
+  }
+}
+)EOF");
+  doRequestCompression({{":method", "post"}}, false);
+  Http::TestResponseHeaderMapImpl headers{{":status", "200"}};
+  doResponseNoCompression(headers);
```

Comment:
your call but I'd be inclined to combine the request and response to 
CompressNoContentLength